spread.yaml

```diff
@@ -125,9 +126,48 @@ prepare: |
   cd "$SPREAD_PATH"
   snap install --classic concierge
 
+
   # Install charmcraft & pipx (on lxd-vm backend)
   concierge prepare --trace
 
+  if [[ -n "$DOCKERHUB_MIRROR" ]] && ([[ $SPREAD_TASK == *"oauth"* ]])
```

Comment:
I believe it is better to have a separate `if (spread_task = *"oauth"` for cases where we are not running in self hosted runners. So we check if the job is oauth, then inside the check we verify if we have dockerhub_mirror and configure it.